[ktx] build failure

**Host Environment**
- OS: Windows
- Compiler: 2022

**To Reproduce**
Using a simple CMake project with the tool chain set to vcpkg's cmake file. With a simple vcpkg.json as follows to include the library:

```json
{
  "name": "name",
  "version-string": "0",
  "dependencies": [
    "ktx"
  ]
}
```

**Failure logs**
```
1> [CMake] -- Running vcpkg install - done
1> [CMake] CMake Error at out/build/x64-debug/vcpkg_installed/x64-windows/share/ktx/KtxTargets.cmake:80 (message):
1> [CMake]   The imported target "KTX::ktx" references the file
1> [CMake] 
1> [CMake]      "C:/Users/mk/source/repos/CMakeProject1/out/build/x64-debug/vcpkg_installed/x64-windows/debug/bin/ktx.dll"
1> [CMake] 
1> [CMake]   but this file does not exist.  Possible reasons include:
1> [CMake] 
1> [CMake]   * The file was deleted, renamed, or moved to another location.
1> [CMake] 
1> [CMake]   * An install or uninstall procedure did not complete successfully.
1> [CMake] 
1> [CMake]   * The installation package was faulty and contained
1> [CMake] 
1> [CMake]      "C:/Users/mk/source/repos/CMakeProject1/out/build/x64-debug/vcpkg_installed/x64-windows/share/ktx/KtxTargets.cmake"
1> [CMake] 
1> [CMake]   but not all the files it references.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   out/build/x64-debug/vcpkg_installed/x64-windows/share/ktx/KtxConfig.cmake:7 (include)
1> [CMake]   C:/src/vcpkg/scripts/buildsystems/vcpkg.cmake:807 (_find_package)
1> [CMake]   CMakeProject1/CMakeLists.txt:9 (find_package)
```

**Additional context**

It appears the .dll file is missing or not being built for ktx. It does build the zstd dll file though, and it does produce a .lib file. Just missing that ktx.dll library file.

--- a/ports/ktx/portfile.cmake
+++ b/ports/ktx/portfile.cmake
@@ -21,7 +21,7 @@ if(VCPKG_TARGET_IS_WINDOWS)
             "https://repo.msys2.org/msys/x86_64/util-linux-2.35.2-1-x86_64.pkg.tar.zst"
             ff951c2cd96d0fda87bacb505c93e4aa1f9aeb35f829c52b5a7862d05e167f69605a4927a0e7197b5ee2b2fa5cb56619ad7a6ba293ede4765fdcacedf2ed35da
         )
-    vcpkg_add_to_path(${MSYS_ROOT}/usr/bin)
+    vcpkg_add_to_path("${MSYS_ROOT}/usr/bin")
 
     file(REMOVE
         "${SOURCE_PATH}/other_include/zstd.h"
@@ -36,19 +36,18 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         vulkan KTX_FEATURE_VULKAN
 )
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         -DKTX_VERSION_FULL=v${PORT_VERSION}
         -DKTX_FEATURE_TESTS=OFF
         -DKTX_FEATURE_LOADTEST_APPS=OFF
         -DKTX_FEATURE_STATIC_LIBRARY=${ENABLE_STATIC}
         ${FEATURE_OPTIONS}
+    DISABLE_PARALLEL_CONFIGURE
 )
 
-vcpkg_install_cmake()
-
+vcpkg_cmake_install()
 vcpkg_copy_pdbs()
 
 if(tools IN_LIST FEATURES)
@@ -64,11 +63,14 @@ if(tools IN_LIST FEATURES)
     vcpkg_copy_tool_dependencies(${CURRENT_PACKAGES_DIR}/tools/${PORT})
 endif()
 
-vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/ktx)
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/ktx)
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/bin")
+
+if("${VCPKG_LIBRARY_LINKAGE}" STREQUAL "static")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/bin")
+endif()
 
 configure_file("${SOURCE_PATH}/LICENSE.md" "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" COPYONLY)
 file(GLOB LICENSE_FILES "${SOURCE_PATH}/LICENSES/*")

--- a/ports/ktx/vcpkg.json
+++ b/ports/ktx/vcpkg.json
@@ -1,11 +1,19 @@
 {
   "name": "ktx",
   "version": "4.0.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "A small library of functions for writing and reading KTX files, and instantiating OpenGL®, OpenGL ES™️ and Vulkan® textures from them",
   "homepage": "https://github.com/KhronosGroup/KTX-Software",
   "supports": "!(uwp | x86 | arm64)",
   "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    },
     "zstd"
   ],
   "features": {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3266,7 +3266,7 @@
     },
     "ktx": {
       "baseline": "4.0.0",
-      "port-version": 1
+      "port-version": 2
     },
     "kubazip": {
       "baseline": "0.1.19",

--- a/versions/k-/ktx.json
+++ b/versions/k-/ktx.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "048534e1a90a5e15ebf065ccfc752c9c5fea77c5",
+      "version": "4.0.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "2d554178c6665e285b8a480dabe83fcf72aa5514",
       "version": "4.0.0",
       "port-version": 1

